wgpu-hal/src/metal/surface.rs

```diff
@@ -34,87 +23,16 @@ impl super::Surface {
         }
     }
 
-    /// If not called on the main thread, this will panic.
-    #[allow(clippy::transmute_ptr_to_ref)]
-    pub unsafe fn from_view(view: NonNull<Object>) -> Self {
-        let layer = unsafe { Self::get_metal_layer(view) };
-        let layer = ManuallyDrop::new(layer);
-        // SAFETY: The layer is an initialized instance of `CAMetalLayer`, and
-        // we transfer the retain count to `MetalLayer` using `ManuallyDrop`.
-        let layer = unsafe { metal::MetalLayer::from_ptr(layer.cast()) };
-        Self::new(layer)
-    }
```

Comment:
question: Pinging @gfx-rs/wgpu here: is it problematic if `metal::Surface::from_view` goes away WRT user-exposed functionality?

Reply:
Yes, it is likely to be an issue

Reply:
Ah, this is public API, didn't realize that since it isn't present on docs.rs.

I'll see what I can do about keeping some form of it around.

Reply:
Have opened https://github.com/gfx-rs/wgpu/pull/7649 for that, will rebase once that lands.